### ISSUE TITLE
Web - Workspace - Workspace settings and chat on the background instead of #admins room

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#18351]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/17447

## Action Performed:
1. Create a new free policy in OldDot
2. User is redirected to NewDot

## Expected Result:
Verify that you're redirected to the Workspace #admins room in NewDot

## Actual Result:
Workspace settings and Workspace chat on the background instead of #admins room

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.9.14

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/235985414-9365ab99-ee60-4673-8292-2688e0a325d4.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01f1b02d43bf434212

**Comment by MelvinBot** [comment]
Current assignee @NicMendonca is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @techievivek (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by alexrasla** [comment]
Hello, how do I launch the Old Dot Expensify app on the web?

**Comment by mollfpr** [comment]
@alexrasla [Staging](https://staging.expensify.com/) and [production](https://www.expensify.com/).

**Comment by alexrasla** [comment]
Can I launch it on localhost?

**Comment by mollfpr** [comment]
Unfortunately not. Currently, only New Dot is open for contributors.

**Comment by alexrasla** [comment]
I see. Do you have suggestions on how to debug and test this issue? Is there an API call that I can make that redirects to the New Expensify page and simulates the issue?

**Comment by mollfpr** [comment]
I usually copy the URL right after being redirected from the OldDot staging and change the domain to localhost.

**Comment by techievivek** [comment]
Not overdue, waiting for @mollfpr to have a look at the proposal.

**Comment by mollfpr** [comment]
I agree with @hoangzinh [proposal](https://github.com/Expensify/App/issues/18351#issuecomment-1536381859). We always navigate the user to workspace settings after the creation. But in this case, we have the `openOnAdminRoom` params where it should be open in the admin room.

I tried the proposal, and it's not navigating the user to the workspace settings page but randomly opening an admin room from another workspace created before.

@hoangzinh Could you address the above issue if it's correct?

**Comment by mollfpr** [comment]
> Just wanna confirm, are we going to change to open admin room of newly created workspace?

I'm unsure what you mean, but we should leave what we have in `openOnAdminRoom` in the `findLastAccessedReport` method.

The ideal solution will be to navigate the user to the workspace admin room instead of to the workspace initial settings page here

https://github.com/Expensify/App/blob/983da48182e158ee5a3df67c04f7d21c8367a2d3/src/libs/actions/Policy.js#L1063-L1066

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mollfpr** [comment]
@mvtglobally I'm still able to reproduce it.


https://user-images.githubusercontent.com/25520267/236870053-32c6f8ae-ad99-4875-b901-89a15baa131f.mov

**Comment by mvtglobally** [comment]
Thanks for confirming @mollfpr. We run validations weekly and sometimes it possible our team is not able to repro the issue as it may be device or platform specific.

**Comment by NicMendonca** [comment]
@mollfpr @techievivek any update here?

**Comment by techievivek** [comment]
Sorry for the delay. @mollfpr You are redirected to the #admin room in the above video, so that is the expected behaviour, right? I wasn't able to reproduce this.

**Comment by techievivek** [comment]
Also, the expected behaviour here is that the user should be redirected to the #admin room with workspace settings RHP open, right?

**Comment by mollfpr** [comment]
@techievivek Yes, it's redirected to #admin room but not the current workspace and it shouldn't open the workspace initial page I guess.

**Comment by techievivek** [comment]
> Yes, it's redirected to the #admin room but not the current workspace

The admin room is from your workspace 2 so I think this is correct.

>  it shouldn't open the workspace initial page I guess.

Sorry, did you mean the settings panel or the workspace room?

**Comment by mollfpr** [comment]
> The admin room is from your workspace 2 so I think this is correct.

That's an old workspace, and it's archived.

> Sorry, did you mean the settings panel or the workspace room?

I mean the settings panel.

**Comment by techievivek** [comment]
@mollfpr Ok, yeah, I see the issue now. I am able to reproduce this locally on DEV. Let me confirm the expected behaviour here with the team before we move with a solution. Thanks

**Comment by techievivek** [comment]
Just had a [discussion](https://expensify.slack.com/archives/C03U7DCU4/p1683789859387059) with the team and this is what we expect to happen when we create a new workspace on oldDot.
1. User should be redirected to oldDot
2. We want to show the workspace settings RHP and the corresponding admin in the background

**Comment by techievivek** [comment]
@hoangzinh Can you please attach a video with your proposal, I am curious to see how does the fix looks in action.

**Comment by mollfpr** [comment]
@hoangzinh Could you update your proposal based on [this](https://github.com/Expensify/App/issues/18351#issuecomment-1544476487) expected result? Thanks!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by techievivek** [comment]
@hoangzinh 
I am not sure if I am fully understanding the RCA part of the proposal here so I have a few questions for you.

From your proposal above 
> When we call the action Policy.createWorkspace to create workspace, in the end, it will always navigate to new created workspace.

Isn't that expected that we want to show the newly created workspace route? This would ensure the RHP gets shown or does it also renders the workspace room in the background(I don't think that is the case, correct me if I am wrong here).

> > The general idea is we check if we're transitioning from oldDot to newDot and openOnAdminRoom param is true, we need to navigate to report admin room before we navigate to new created workspace. So it will show new workspace in RHN and corresponding admin room in the background.

> To do this, I think we can pass an additional openAdminRoom param to action Policy.createWorkspace

So if it is a transitioning action then we add an extra param `openOnAdminRoom` to make sure we end up in the admin room, am I understanding that correctly?

**Comment by techievivek** [comment]
@mollfpr Can you also have a look at the updated proposal? Thanks

**Comment by mollfpr** [comment]
Sorry for the delay 🙏

>Then before this [line](https://github.com/Expensify/App/blob/ab9171f953603bb8e4e061f5582a4b7cab3d4a43/src/libs/actions/Policy.js#L1066), we have to check if openAdminRoom is true, we need navigate to this [adminsChatReportID created here](https://github.com/Expensify/App/blob/ab9171f953603bb8e4e061f5582a4b7cab3d4a43/src/libs/actions/Policy.js#L864).
Navigation.navigate(ROUTES.getReportRoute(adminsChatReportID))

We should not navigate to the report before navigating again to the workspace settings page. Before navigating to the workspace settings page, we are already on the report page, so we need to fix the `reportID` set to initial params with the correct `reportID` for the workspace.

**Comment by mollfpr** [comment]
>Then before this [line](https://github.com/Expensify/App/blob/ab9171f953603bb8e4e061f5582a4b7cab3d4a43/src/libs/actions/Policy.js#L1066), we have to check if openAdminRoom is true, we need navigate to this [adminsChatReportID created here](https://github.com/Expensify/App/blob/ab9171f953603bb8e4e061f5582a4b7cab3d4a43/src/libs/actions/Policy.js#L864).
Navigation.navigate(ROUTES.getReportRoute(adminsChatReportID))

Also, I got this console error when trying the above solution.

<img width="656" alt="Screenshot 2023-05-17 at 14 15 44" src="https://github.com/Expensify/App/assets/25520267/7a6ba6fc-410a-47e9-a3cd-082748752c22">

**Comment by mollfpr** [comment]
>@mollfpr To fix above error. I already mentioned here

Sorry, I didn't see that. Are we going to redirect the user twice to admin room on `openOnAdminRoom` and with your solution `Navigation.navigate(ROUTES.getReportRoute(adminsChatReportID))`?

>btw @mollfpr do you think this https://github.com/Expensify/App/issues/18894 is almost dup with our issue here. If yes, we can select proposals there for more options

Good point. I think yes and no, I'm still incline to the option of `openOnAdminRoom` when creating workspace. I seem that will be an option to open the admin room or not.

**Comment by mollfpr** [comment]
@techievivek @NicMendonca What are your thoughts on merging this issue with https://github.com/Expensify/App/issues/18894?

**Comment by techievivek** [comment]
@mollfpr Interesting, but that looks like a bug when you create the workspace right from the newDot and this seems to be happening when you create the workspace from the oldDot. Is the redirection part of the logic same in both cases? Maybe we can put a HOLD on this issue and re-visit this when the linked PR is merged?

**Comment by mollfpr** [comment]
@techievivek https://github.com/Expensify/App/pull/19224 is merged to staging.

It's still set to the wrong admin room report in the background.

https://github.com/Expensify/App/assets/25520267/fb22ea9c-0820-4b4b-b5ad-a178d612d313

**Comment by techievivek** [comment]
@mollfpr I would say that is expected, no? Because that fix was when we created the workspace right from newDot, but this issue seems to happen when we get redirected from the oldDot after creating a workspace. Can you confirm if @hoangzinh's proposal will fix the issue here? Thanks

**Comment by NicMendonca** [comment]
Going OOO until June 5th so assigning a buddy to this GH to watch over the issue. I'll pick it back up when I am back from OOO if this is still open.

**Comment by mollfpr** [comment]
@techievivek latest [proposal](https://github.com/Expensify/App/issues/18351#issuecomment-1536381859) from @hoangzinh seems working great! Let's go with their proposal.

🎀 👀 🎀 C+ reviewed!

**Comment by situchan** [comment]
Drawer navigation will be soon deprecated from navigation reboot.
If solution is related to `Navigation.isDrawerReady`, we can put this on hold.

**Comment by mollfpr** [comment]
Thanks for the suggestion @situchan! I just checked the latest PR for navigation reboot https://github.com/Expensify/App/pull/19263, and the issue remains even with the removal of `Navigation.isDrawerReady`.

https://github.com/Expensify/App/assets/25520267/4d4429d7-98eb-4ba4-8f20-c4685068fc57

The navigation history is still not working well on the current refactor PR, and there's much more to check with the navigation reboot transition.

@techievivek Should we hold this issue until the navigation refactor is done? I think it's better if we can solve it now and propose a regression step afterward.

**Comment by conorpendergrast** [comment]
@techievivek Can you confirm @mollfpr's question [above](https://github.com/Expensify/App/issues/18351#issuecomment-1560543634)?

**Comment by techievivek** [comment]
Yes, I agree that it would be better to tackle this once the navigation refactor is done. Adding the HOLD for now. Thanks

**Comment by NicMendonca** [comment]
@conorpendergrast I am back so unassigning you 😃

**Comment by techievivek** [comment]
Looks like the main navigation refactor PR is merged, so I am removing the HOLD now to get this going. @hoangzinh, Can you please re-look into your proposal and see if anything needs to be updated? Please bump us when it's ready for review. Thanks

**Comment by techievivek** [comment]
@mollfpr, Will you please take a look at the updated proposal and review it? If everything looks good, then we can go forward with the proposal. Thanks

**Comment by mollfpr** [comment]
Sorry for the delay. @techievivek the proposal looks good to me 👍

🎀 👀 🎀 C+ reviewed!

**Comment by NicMendonca** [comment]
@jliexpensify I am going OOO until Wednesday. Can you watch this while I am away? I'll unassign you when I am back. Thank you!

**Comment by techievivek** [comment]
@hoangzinh Thanks for the proposal, it looks good to me. You can now start working on the PR.

**Comment by NicMendonca** [comment]
PR is making progress: https://github.com/Expensify/App/pull/21572#issuecomment-1614804576 ✅

**Comment by NicMendonca** [comment]
PR update: https://github.com/Expensify/App/pull/21572#issuecomment-1622038968

**Comment by mollfpr** [comment]
> [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
 [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

No offending PR

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

This is a regression from the Navigation reboot, so the regression test should be enough.

>[@mollfpr] Determine if we should create a regression test for this bug.
 [@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**_On Web_**
1. Login to NewDot
2. Login with same account to OldDot
3. Create a new free policy in OldDot
4. Verify that you are redirected to the #admin room with new workspace settings RHP open in NewDot
5. Verify that there is no infinite loading overlay

**_On mWeb_**
1. Login to OldDot
2. Create a new free policy in OldDot
3. Verify that you are in new workspace settings in NewDot

**Comment by NicMendonca** [comment]
Sorry for the delay!

**Comment by NicMendonca** [comment]
@hoangzinh sent you the job offer: https://www.upwork.com/jobs/~01e880185a86822464

@mollfpr just to confirm, you've requested payment via Expensify?

**Comment by mollfpr** [comment]
@NicMendonca Nope. Still using Upwork 😅

**Comment by NicMendonca** [comment]
ah, opps! @mollfpr then you'll want to accept the job offer also: https://www.upwork.com/jobs/~01e880185a86822464

**Comment by mollfpr** [comment]
@NicMendonca I just apply to the job, thank you!

**Comment by NicMendonca** [comment]
everyone paid ✅ BZ checlikst ✅ 

thanks everyone!
